File: C#/Ops/ParticipantInfoData.cs
///////////////////////////////////////////////////////////
//  ParticipantInfoData.cs
//  Implementation of the Class ParticipantInfoData
//  Created on:      01-jan-2013 
//  Author:
///////////////////////////////////////////////////////////

using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace Ops
{
    // NOTE. Must be kept in sync with C++
	public class ParticipantInfoData : OPSObject
	{
        public string name { get; set; }
        public string id { get; set; }
        public string domain { get; set; }
        public string ip { get; set; }
        public string languageImplementation { get; set; }
        public string opsVersion { get; set; }
        public int mc_udp_port { get; set; }
        public int mc_tcp_port { get; set; }

		private List<TopicInfoData> _subscribeTopics = new List<TopicInfoData>();
        public List<TopicInfoData> subscribeTopics { get { return _subscribeTopics; } set { _subscribeTopics = value; } }

        private List<TopicInfoData> _publishTopics = new List<TopicInfoData>();
        public List<TopicInfoData> publishTopics { get { return _publishTopics; } set { _publishTopics = value; } }

        // C++ version: std::vector<string> knownTypes;
        private List<string> _knownTypes = new List<string>();
        [Editor(@"System.Windows.Forms.Design.StringCollectionEditor," +
            "System.Design, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
            typeof(System.Drawing.Design.UITypeEditor))]
        public List<string> knownTypes { get { return _knownTypes; } set { _knownTypes = value; } }

		
		public ParticipantInfoData()
		{
			AppendType("ops.ParticipantInfoData");
		}

        public override void Serialize(IArchiverInOut archive)
		{
			base.Serialize(archive);

			name = archive.Inout("name", name);
			domain = archive.Inout("domain", domain);
			id = archive.Inout("id", id);
			ip = archive.Inout("ip", ip);
			languageImplementation = archive.Inout("languageImplementation", languageImplementation);
			opsVersion = archive.Inout("opsVersion", opsVersion);
			mc_udp_port = archive.Inout("mc_udp_port", mc_udp_port);
			mc_tcp_port = archive.Inout("mc_tcp_port", mc_tcp_port);
            _subscribeTopics = (List<TopicInfoData>)archive.InoutSerializableList<TopicInfoData>("subscribeTopics", _subscribeTopics);
            _publishTopics = (List<TopicInfoData>)archive.InoutSerializableList<TopicInfoData>("publishTopics", _publishTopics);
            _knownTypes = (List<string>)archive.InoutStringList("knownTypes", _knownTypes);
		}
	};

}

File: Java/OPSJLib/src/ops/ParticipantInfoData.java
///////////////////////////////////////////////////////////
//  ParticipantInfoData.java
//  Implementation of the Class ParticipantInfoData
//  Created on:      10-jan-2013
//  Author:
///////////////////////////////////////////////////////////

package ops;

import ops.OPSObject;
import configlib.ArchiverInOut;
import configlib.SerializableFactory;
import configlib.Serializable;
import java.io.IOException;

// NOTE. Must be kept in sync with C++
public class ParticipantInfoData extends OPSObject
{
    public String name;
    public String id;
    public String domain;
    public String ip;
    public String languageImplementation;
    public String opsVersion;
    public int mc_udp_port;
    public int mc_tcp_port;

    public java.util.Vector<TopicInfoData> subscribeTopics = new java.util.Vector<TopicInfoData>();

    public java.util.Vector<TopicInfoData> publishTopics = new java.util.Vector<TopicInfoData>();

    // C++ version: std::vector<string> knownTypes;
    public java.util.Vector<String> knownTypes = new java.util.Vector<String>();

		
    public ParticipantInfoData()
    {
        super();
	appendType("ops.ParticipantInfoData");
    }

    public void serialize(ArchiverInOut archive) throws IOException
    {
	super.serialize(archive);

	name = archive.inout("name", name);
	domain = archive.inout("domain", domain);
	id = archive.inout("id", id);
	ip = archive.inout("ip", ip);
	languageImplementation = archive.inout("languageImplementation", languageImplementation);
	opsVersion = archive.inout("opsVersion", opsVersion);
	mc_udp_port = archive.inout("mc_udp_port", mc_udp_port);
	mc_tcp_port = archive.inout("mc_tcp_port", mc_tcp_port);
        /// The following requires that TopicInfoData can be created by the OPSObjectFactory. Differs from C++ & C# that has a different solution.
        subscribeTopics = (java.util.Vector<TopicInfoData>)archive.inoutSerializableList("subscribeTopics", subscribeTopics);
        publishTopics = (java.util.Vector<TopicInfoData>)archive.inoutSerializableList("publishTopics", publishTopics);
	knownTypes = (java.util.Vector<String>) archive.inoutStringList("knownTypes", knownTypes);
    }

}

